middleware/transactionBreakdownDataValidation.ts

```diff
@@ -0,0 +1,32 @@
+import Joi, { ObjectSchema } from 'joi';
+import { NextFunction, Request, Response } from 'express';
+
+export const transactionBreakdownDataValidation = (schema: ObjectSchema) => {
```

Comment:
We can create a general `RequestInputSchemaValidator` class and put this in a `validate` method that takes in one param: the schema. We can put the `transactionBreakdownSchema` in another file within this folder (maybe create another folder called `RequestInputSchemas` or smthg. 

The reason being is there is actually nothing unique to `transactionBreakdownData` with this method. This method is very general and can be re-used for validating inputs in other endpoints so we should separate wherever is appropriate.

Reply:
I think i got what you mean, I ended up separating out the schema and turning `transactionBreakdownSchema` into `RequestInputSchemaValidator `. I'm not entirely sure what you mean by making a class with a method when RequestInputSchemaValidator is already a method, but we can chat on this if anything

Reply:
@elliely32 you didn't have RequestInputSchemaValidator when I reviewed this haha. [See reviewed changes](https://github.com/tally-team/tally-backend/pull/69/files/459744640eb60fcbf8b0f7c4d1ebf5ab9aeace47).

Reply:
@michello oh yea that's new, it was originally `transactionBreakdownDataValidation `